style_editor.html

```diff
@@ -5671,18 +5671,21 @@
 
 function Saw(RPM, LOW, HIGH) { return new SawClass(RPM, LOW, HIGH); }
 
-const  TRIGGER_ATTACK = 0;
-const  TRIGGER_SUSTAIN = 1;
-const  TRIGGER_RELEASE = 2;
-const  TRIGGER_OFF = 3;
+const  TRIGGER_DELAY = 0;
+const  TRIGGER_ATTACK = 1;
+const  TRIGGER_SUSTAIN = 2;
+const  TRIGGER_RELEASE = 3;
+const  TRIGGER_OFF = 4;
 
 class TriggerClass extends FUNCTION {
-  constructor(EFFECT, FADE_IN_MILLIS, SUSTAIN_MILLIS, FADE_OUT_MILLIS) {
-    super("When EFFECT occors, ramps up to 32768, stays there for SUSTAIN_MILLIS, then ramps down again.", arguments);
+  constructor(EFFECT, FADE_IN_MILLIS, SUSTAIN_MILLIS, FADE_OUT_MILLIS, DELAY) {
+    super("When EFFECT occurs, DELAY controls a pause, then we ramp up to 32768, stay there for SUSTAIN_MILLIS, then ramp down again.", arguments);
     this.add_arg("EFFECT", "EFFECT", "Trigger event");
     this.add_arg("FADE_IN_MILLIS", "FUNCTION", "How long it takes to ramp to 32768");
     this.add_arg("SUSTAIN_MILLIS", "FUNCTION", "Stay at 32768 for this long.");
     this.add_arg("FADE_OUT_MILLIS", "FUNCTION", "How long it takes to ramp back down to zero.");
+    this.add_arg("DELAY_MILLIS", "FUNCTION", "How long to delay before trigger starts.");
```

Comment:
Shouldn't this have a default of zero?

Reply:
It sure should.

Reply:
fixed

Reply:
I still don't see a default n this call to add_arg()

Reply:
Sorry, I'm not sure on this.   
  I thought the default you meant was in the style template
  `AddFunction("Trigger<EFFECT_FORCE, Int<500>, Int<1000>, Int<500>, Int<0>>");`
  Do you mean something like this?
`class TriggerClass extends FUNCTION {
  constructor(EFFECT, FADE_IN_MILLIS, SUSTAIN_MILLIS, FADE_OUT_MILLIS, DELAY_MILLIS = 0) {`
  That's my best guess based on the C++ code translation from:
  ```
  template<
  BladeEffectType EFFECT,
  class FADE_IN_MILLIS,
  class SUSTAIN_MILLIS,
  class FADE_OUT_MILLIS,
  class DELAY_MILLIS = Int<0>>
  ```
  
  It works as expected with or without this so as I said I'm not sure.

Reply:
oh. is it this?
` this.add_arg("DELAY_MILLIS", "FUNCTION", "How long to delay before trigger starts.", Int(0));`

Reply:
Added in latest commit.